### ISSUE TITLE
Feature Request: configurable emoji setups

### ISSUE THREAD

**Issue by jhechtf** [benjaminadk/emojigit#8]
Greetings,

Would it be possible to add in configuration map for the emojis used? Not everyone is going to agree with the emojis that were chosen, and in my looking it doesn't appear that there is a way to alter the emojis without altering some of your code by hand (which is not the best of ideas).

I figure if the mapping of purpose-> emoji was a setting that could be extended or altered by the user then teams could create their own agreed-upon map and simply share it between them through their onboarding docs or what have you. I have some experience with doing this if you would like some help.

**Comment by benjaminadk** [comment]
This is an interesting idea. When I originally put this together the team I was working with was going by [this emoji config](https://gitmoji.carloscuesta.me/). I am not aware of any other "standards" out there, but can easily see teams/implementing there own.

I guess there could be an option where the user supplies there own mapping. Currently, there are three properties on each gitmoji:  `{emoji: '✨', colon: ':sparkles:', text: 'Initial commit'}`. So I don't see why there couldn't be an option/s to extend or replace the current mapping. My first thought is to have the user add their custom array of emoji objects in the VSCode settings. Then the extension would check this option and either use the original or custom list in the UI.

I think I can add this pretty easily. Any thoughts on this approach?

**Comment by benjaminadk** [comment]
https://github.com/benjaminadk/emojigit/commit/7ef66885d6d75c73f95f2edc8ab2d20b91c36f9e

I added the feature as I described. Things to note:
- as of now there is no validation for custom map. Extension will just crash if an object does not have required properties. When i get some more time i'll go in and touch things up.
- text describing commit type in the quickpick will be cut off at 60 characters.
